gtsam/nonlinear/NonlinearOptimizerParams.h

```diff
@@ -81,10 +103,16 @@ class GTSAM_EXPORT NonlinearOptimizerParams {
     CHOLMOD, /* Experimental Flag */
   };
 
-  LinearSolverType linearSolverType; ///< The type of linear solver to use in the nonlinear optimizer
+  LinearSolverType linearSolverType = MULTIFRONTAL_CHOLESKY; ///< The type of linear solver to use in the nonlinear optimizer
   boost::optional<Ordering> ordering; ///< The optional variable elimination ordering, or empty to use COLAMD (default: empty)
   IterativeOptimizationParameters::shared_ptr iterativeParams; ///< The container for iterativeOptimization parameters. used in CG Solvers.
 
+  NonlinearOptimizerParams() = default;
```

Comment:
I remember seeing a comment from @dellaert about having variables before constructors, so he'll be happy about this one. :smile: